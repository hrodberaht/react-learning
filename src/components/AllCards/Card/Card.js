import React, { Component } from 'react';
import PropTypes from 'prop-types';

export default class Card extends Component {
  card = {
    border: '1px solid black',
    width: '300px',
    margin: '5px auto',
  };

  button = {
    color: 'white',
    border: '1px solid red',
    fontSize: '16px',
    backgroundColor: 'red',
    padding: '5px',
  };

  render() {
    const {
      handleClick,
      card: { name, email, id },
    } = this.props;
    return (
      <div style={this.card}>
        <h2>
          Name:
          {name}
        </h2>
        <h4>
          Email:
          {email}
        </h4>
        <button
          style={this.button}
          type="button"
          onClick={() => handleClick(id)}
        >
          Delete
        </button>
      </div>
    );
  }
}

Card.propTypes = {
  handleClick: PropTypes.func.isRequired,
  card: PropTypes.shape(
    {
      name: PropTypes.string,
      email: PropTypes.string,
      id: PropTypes.number,
    },
  ).isRequired,
};
