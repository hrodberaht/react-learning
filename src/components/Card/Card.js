import React, { Component } from "react";

export default class Card extends Component {
  render() {
    const { name, email } = this.props.card;
    return (
      <div>
        <h2>Name: {name} </h2>
        <h4>Email: {email}</h4>
      </div>
    );
  }
}
