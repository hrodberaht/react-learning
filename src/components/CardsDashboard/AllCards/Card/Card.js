import React, { Component } from 'react';
import PropTypes from 'prop-types';

export default class Card extends Component {
  render() {
    const {
      handleClick,
      card: { name, email, id },
    } = this.props;
    return (
      <div className="cards-list__card">
        <p>Name:</p>
        <h2>{name}</h2>
        <p>Email:</p>
        <h4>{email}</h4>
        <button
          className="btn danger-btn"
          type="button"
          onClick={() => handleClick(id)}
        >
          Delete
        </button>
      </div>
    );
  }
}

Card.propTypes = {
  handleClick: PropTypes.func.isRequired,
  card: PropTypes.shape(
    {
      name: PropTypes.string,
      email: PropTypes.string,
      id: PropTypes.number,
    },
  ).isRequired,
};
