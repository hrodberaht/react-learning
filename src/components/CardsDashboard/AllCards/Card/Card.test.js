import React from 'react';
import { shallow } from 'enzyme';
import Card from './Card';

describe('<Card />', () => {
  let wrapper;
  const props = {
    card: {
      id: 1,
      name: 'John',
      email: 'john@gmail.com',
    },
    handleClick: jest.fn(),
  };


  beforeEach(() => {
    wrapper = shallow(<Card {...props} />);
  });

  it('should render component witout crash', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should run function after click button', () => {
    wrapper.find('button').simulate('click');

    expect(props.handleClick.mock.calls.length).toBe(1);
  });
});
