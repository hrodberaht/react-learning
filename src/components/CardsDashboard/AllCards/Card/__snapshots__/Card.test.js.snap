// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> should render component witout crash 1`] = `
<div
  style={
    Object {
      "border": "1px solid black",
      "margin": "5px auto",
      "width": "300px",
    }
  }
>
  <h2>
    Name:
    John
  </h2>
  <h4>
    Email:
    john@gmail.com
  </h4>
  <button
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "red",
        "border": "1px solid red",
        "color": "white",
        "fontSize": "16px",
        "padding": "5px",
      }
    }
    type="button"
  >
    Delete
  </button>
</div>
`;
